include/deal.II/lac/superlu_mt.h

```diff
@@ -0,0 +1,190 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2001 - 2020 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+#ifndef dealii_superlu_mt_h
+#define dealii_superlu_mt_h
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/exceptions.h>
+#include <deal.II/base/subscriptor.h>
+
+#include <deal.II/lac/sparse_matrix.h>
+#include <deal.II/lac/vector.h>
+
+#include "slu_mt_ddefs.h"
+//#include "slu_mt_sdefs.h"
+
+#include <vector>
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * Wrapper for SuperLU-MT direct linear solver.
+ * \todo Complete the docu here.
+ * *
+ * @ingroup Solvers Preconditioners
+ *
+ * @author Paras Kumar, 2020
+ */
+class SuperLU_MT : public Subscriptor
+{
+public:
+  /**
+   * Declare type for container sizes.
+   */
+  using size_type = types::global_dof_index;
+
+
+  // \todo remove this issue below
+  /**
+   * in order to avoid the error with
+   * <c> invalid conversion from ‘unsigned int*’ to ‘int_t* {aka int*}’ </c>
+   * we create a temporary type. Ideally this should be same as \c size_type
+   */
+  using my_int = int_t;
+
+  /**
+   * Declare types for matrix and vectorm for extension to block sparse matrices
+   * (to be replaced by instantiation later)
+   */
+  using number_type = double;
+  using matrix_type = SparseMatrix<number_type>;
+  using vector_type = Vector<number_type>;
+
+  /**
+   * To contain additional data such as number of process for multi-threading.
+   */
+  struct AdditionalData //: public superlumt_options_t
+  {
+  public:
+    AdditionalData()
+      : column_ordering(0)
+    {
+      options_.nprocs    = MultithreadInfo::n_threads();
+      options_.fact      = DOFACT;
+      options_.PrintStat = NO;
+    }
+
+    explicit AdditionalData(const unsigned int &num_threads,
+                            const unsigned int &factorize,
+                            const size_type &   col_order,
+                            const bool          printSolverStats)
+      : column_ordering(col_order)
+    {
+      options_.nprocs = num_threads;
+      options_.fact   = static_cast<fact_t>(factorize);
+      if (printSolverStats)
+        options_.PrintStat = YES;
+      else
+        options_.PrintStat = NO;
+    }
+    /**
+     * Object to store additional options for the solver
+     */
+    superlumt_options_t options_;
```

Comment:
Since you reference SuperLU data types and functions in this file, you will have to put most of it into `#ifdef DEAL_II_WITH_SUPERLU`.